apps/bfd-model/bfd-model-codebook-plugin/src/test/java/gov/cms/bfd/model/codebook/extractor/PdfParserTest.java

```diff
@@ -30,18 +36,118 @@
+          variablesById.put(variable.getId(), new ArrayList<>());
+
+        variablesById.get(variable.getId()).add(variable);
+      }
+    }
+
+    // Find all of the variables that appear in more than one Codebook.
+    List<String> duplicatedVariableIds = new ArrayList<>();
+    for (String variableId : variablesById.keySet()) {
+      List<Variable> variables = variablesById.get(variableId);
+      if (variables.size() > 1) duplicatedVariableIds.add(variableId);
+    }
+
+    // Log a detailed warning for each duplicate.
+    for (String duplicatedVariableId : duplicatedVariableIds) {
+      List<Variable> duplicatedVariables = variablesById.get(duplicatedVariableId);
+      LOGGER.warn(
+          "The variable '{}' appears more than once: {}.",
+          duplicatedVariableId,
+          duplicatedVariables);
+    }
+
+    /*
+     * We know that these variables are duplicated. It isn't great, but oh well. We
+     * assert their presence just to get alerted if something interesting changes
+     * with the data or parsing.
+     */
+    assertTrue(duplicatedVariableIds.contains("BENE_ID"));
+    assertTrue(duplicatedVariableIds.contains("DOB_DT"));
+    assertTrue(duplicatedVariableIds.contains("GNDR_CD"));
+
+    // Blow up if anything more than those known problems appears.
+    assertEquals(87, duplicatedVariableIds.size());
+  }
+
+  /**
+   * Parses all of the {@link SupportedCodebook}s using {@link
+   * gov.cms.bfd.model.codebook.extractor.PdfParser}, looking for duplicate {@link Value#getCode()}s
+   * within each {@link Variable}.
+   *
+   * @throws IOException Indicates test error.
+   */
+  @Test
+  public void findDuplicateCodes() throws IOException {
+    for (SupportedCodebook supportedCodebook : SupportedCodebook.values()) {
+      Codebook codebook = parser.parseCodebookPdf(supportedCodebook);
+      for (Variable variable : codebook.getVariables()) {
+        if (!variable.getValueGroups().isPresent()) continue;
+
+        // Build a multimap of all the Values by their codes.
+        Map<String, List<Value>> valuesByCode = new LinkedHashMap<>();
+        for (ValueGroup valueGroup : variable.getValueGroups().get()) {
+          for (Value value : valueGroup.getValues()) {
+            if (!valuesByCode.containsKey(value.getCode()))
+              valuesByCode.put(value.getCode(), new ArrayList<>());
+
+            valuesByCode.get(value.getCode()).add(value);
+          }
+        }
+
+        // Find all of the codes that appear in more than one Value.
+        List<String> duplicatedCodes = new ArrayList<>();
+        for (String code : valuesByCode.keySet()) {
+          List<Value> values = valuesByCode.get(code);
+          if (values.size() > 1) duplicatedCodes.add(code);
+        }
+
+        // Log a detailed warning for each duplicate.
+        for (String duplicatedCode : duplicatedCodes) {
+          List<Value> duplicatedValues = valuesByCode.get(duplicatedCode);
+          LOGGER.warn(
```

Comment:
This seems handy for debugging, but do we need this output on every test run? (I assume a warn is going to output with our default log levels)

Reply:
Good question.  That is a direct transfer from `SupportedCodebookTest`.  It logs all of that for me in master as well.  Maven hides the log output from tests so we never noticed it before.  I can see it in idea though.